pkg/resource/internal/compare/compare.go

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package compare implements term operation helpers.
+package compare
+
+import (
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// GetNumbers returns numbers parsed from left and right params, if any of string to number conversion fails returns false as the 3rd arg.
+func GetNumbers(left, right string) (int64, int64, bool) {
+	numLeft, ok := parseValue(left)
+	if !ok {
+		return 0, 0, false
+	}
+
+	numRight, ok := parseValue(right)
+	if !ok {
+		return 0, 0, false
+	}
+
+	return numLeft, numRight, true
+}
+
+func parseValue(value string) (int64, bool) {
+	value = strings.TrimSpace(value)
+
+	sign := int64(1)
+
+	if string(value[0]) == "-" {
+		value = value[1:]
+		sign = -1
+	}
+
+	splitPoint := len(value)
+
+	for i, letter := range value {
+		if !unicode.IsDigit(letter) {
+			splitPoint = i
+
+			break
+		}
+	}
+
+	digits, units := value[:splitPoint], value[splitPoint:]
+
+	if len(digits) == 0 {
+		return 0, false
+	}
+
+	res, err := strconv.ParseInt(digits, 10, 64)
+	if err != nil {
+		return 0, false
+	}
+
+	multiplier, ok := getMultiplier(units)
+	if !ok {
+		return 0, false
+	}
+
+	return res * multiplier * sign, true
+}
+
+func getMultiplierForUnits(value string) (int64, bool) {
+	var val int64
+
+	switch strings.ToLower(value) {
+	case "pi":
+		val = 1 << 50
+	case "ti":
+		val = 1 << 40
+	case "gi":
+		val = 1 << 30
+	case "mi":
+		val = 1 << 20
+	case "ki":
+		val = 1 << 10
+	case "p":
+		val = 1e15
+	case "t":
+		val = 1e12
+	case "g":
+		val = 1e9
+	case "m":
+		val = 1e6
+	case "k":
+		val = 1e3
+	default:
+		return 0, false
+	}
+
+	return val, true
```

Comment:
why? if we have `default:` ?

Reply:
the code above doesn't do return, so this is necessary.
But I'll update the code as you suggested so it'll be irrelevant anyways